### ISSUE TITLE
tf.nn.embedding_lookup_sparse Converting sparse IndexedSlices Warning

### ISSUE THREAD

**Issue by davidenunes** [tensorflow/tensorflow#23566]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): r1.11
- Python version: 3.6
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10/7.3

**Describe the current behavior**
Using embedding_lookup_sparse raises the following warning:

> UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory. "Converting sparse IndexedSlices to a dense Tensor of unknown shape."

Looking at embedding_lookup_sparse implementation, the issue comes from the gather after embedding lookup.

```
...
ids = sp_ids.values
ids, idx = array_ops.unique(ids)

embeddings = embedding_lookup(params,ids,...)
...
embeddings = array_ops.gather(embeddings, idx)
```

**Describe the expected behavior**
I didn't expect the sparse gradients to be passed to an op that can't back-propagate them inside 
embedding lookup sparse.

Is there a reason this op does not call embedding_lookup on sp_ids.values directly? As opposed to
looking up the unique ids and calling gather on those rows ?

**Comment by drpngx** [comment]
It was an optimization to unique ids before sending the RPC calls. It came through [here](https://github.com/tensorflow/tensorflow/commit/415ea7360d3f57249fc18e068852a8b8ce6d7f77#diff-fbcd994b5a5d10595f0307b63f247ff7).

@jhseu any further comment?

**Comment by davidenunes** [comment]
Thank you for looking into it. I suspected it could be something like that, it does have the problem I described though. There's an implicit conversion to a dense tensor, so the gradients are not sparse.

**Comment by georgesterpu** [comment]
I am having the same issue with `tf.nn.embedding_lookup`, snippet below:

```
embedding_matrix = tf.get_variable(
                    name="embedding_matrix",
                    shape=[self._vocab_size, self._embedding_size],  # Python ints, [32x128]
                )

decoder_inputs = tf.nn.embedding_lookup(embedding_matrix, labels)
```

Warning triggered when calling `tf.gradients`.

**Comment by LucaCappelletti94** [comment]
Any development?

**Comment by sushreebarsa** [comment]
@davidenunes We see that you are using old version of tensorflow 1.x which is out of support window, We recommend that you upgrade to 2.6.0  and let us know if the issue still persists in newer versions .Thanks!

**Comment by LucaCappelletti94** [comment]
I am using TensorFlow 2.6 and the warning persists.

**Comment by StarWang** [comment]
Why is this issue closed? We are still seeing the issue

**Comment by LucaCappelletti94** [comment]
@StarWang, please do try to answer the bot form. Hopefully, they will upgrade it, so the bot does not preemptively close issues when no solution is known yet.

**Comment by gadagashwini** [comment]
Hi @davidenunes,
Looks like `tf.nn.embedding_lookup_sparse()` works fine with Tf v2.9. 

```
import tensorflow as tf
params = tf.constant([10.0,20.0])
sp_ids = tf.sparse.SparseTensor(indices=[[0, 0], [1, 2]], values=[0,1], dense_shape=[2, 4])
tf.nn.embedding_lookup_sparse(params, sp_ids,sp_weights=None )
```
**Output**
`<tf.Tensor: shape=(2,), dtype=float32, numpy=array([10., 20.], dtype=float32)>`

```
import tensorflow as tf
params = tf.constant([10.0,20.0, 30.0, 40.0])
sp_ids = tf.constant([0,1,2,3])
tf.nn.embedding_lookup(params,ids)
```
**Output**
`<tf.Tensor: shape=(4,), dtype=float32, numpy=array([10., 20., 30., 40.], dtype=float32)>`

**Comment by LucaCappelletti94** [comment]
What is the minimum version for which this works now? Sadly in my computer science department, there are lots of machines with relatively old GPUs that will not support Tensorflow 2.9. Furthermore, do you have any idea what was the origin of the warning? How was it resolved? Or was it simply silenced?

**Comment by gadagashwini** [comment]
Hi @LucaCappelletti94, Minimum Tensorflow version required is 2.4. Warnings are not just simply silenced, they have been taken care in latest Tf versions. Thank you!

**Comment by LucaCappelletti94** [comment]
I have just tested this in TensorFlow `2.9`, and **the warning**  about the memory usage is still very much present. The method, of course, still works, but that was **never** the issue here.

**Comment by gadagashwini** [comment]
@LucaCappelletti94, Request you to raise new issue by providing all the required details in the template. Thank you!

**Comment by LucaCappelletti94** [comment]
Since it is the same issue it would not make sense to create a new one. It is simply this same one, as reported by @davidenunes, and can be recreated by the code reported in the original issue.

**Comment by SuryanarayanaY** [comment]
Hi @davidenunes ,

This seems resolved already as I don't see any warning now with latest TF version. 
Logs:
```
(tf-metal) suryanarayanay-macbookpro:Downloads suryanarayanay$ python 23566.py
Metal device set to: Apple M1 Pro

systemMemory: 16.00 GB
maxCacheSize: 5.33 GB

tf.Tensor([10. 20.], shape=(2,), dtype=float32)

(tf-metal) suryanarayanay-macbookpro:Downloads suryanarayanay$ cat 23566.py
# -*- coding: utf-8 -*-
"""23566.ipynb

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1nb1ysIMYPWWFpV4_ZoFyse16nHhkzJ0M?resourcekey=0-9tFQh4X0WTIx2P-lSMxsxw
"""

import os
os.environ['TF_CPP_MIN_LOG_LEVEL']='1'

import tensorflow as tf
params = tf.constant([10.0,20.0])
sp_ids = tf.sparse.SparseTensor(indices=[[0, 0], [1, 2]], values=[0,1], dense_shape=[2, 4])
print(tf.nn.embedding_lookup_sparse(params, sp_ids,sp_weights=None ))
```
